pilot/cmd/cmd.go

```diff
@@ -40,3 +42,114 @@ func ReadMeshNetworksConfig(filename string) (*meshconfig.MeshNetworks, error) {
 	}
 	return model.LoadMeshNetworksConfig(string(yaml))
 }
+
+// Validate checks the field values on Cluster with the rules defined in the
+// proto definition for this message. If any rules are violated, an error is returned.
+func ValidateMeshConfig(mesh *meshconfig.MeshConfig) error {
```

Comment:
I strongly suggest moving this validation to model/validation.go - thats the file where we have all the validation code.
secondly, call this ValidateLocalityLBSetting or something to that effect.

Reply:
sorry, just noticed this, would do this in a follow up.